app/models/service_template_transformation_plan_task.rb

```diff
@@ -71,6 +71,15 @@ def preflight_check
     destination_cluster
     virtv2v_disks
     network_mappings
+
+    begin
+      raise unless source.host.verify_credentials?
+    rescue => err
+      msg = "Invalid credentials for '#{source.host.name}'"
+      msg += ": #{err.message}" unless err.message.empty?
```

Comment:
Prefer `<<` over `+=` since you own the String, and also don't need to reallocate a new String.